### ISSUE TITLE
ERROR TypeError: params.then is not a function

### ISSUE THREAD

**Issue by iancward** [jvarness/hubot-lunchtime#14]
Hey there,
I tried using the new code and it looks like it's not happy about how the method is being called:
```
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: [Sun Jul 23 2017 13:46:56 GMT-0700 (PDT)] DEBUG Received message: '@mybot lunchtime' in channel: mybot-testing, from: ian.ward
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: [Sun Jul 23 2017 13:46:56 GMT-0700 (PDT)]
 DEBUG Message '@mybot lunchtime' matched regex //^\s*[@]?mybot[:,]?\s*(?:lunchtime\W*(near (.*) thats (.*)|near (.*)|thats (.*))?)/i/; listener.options = { id: null }
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: [Sun Jul 23 2017 13:46:56 GMT-0700 (PDT)] DEBUG Executing listener callback for Message '@mybot lunchtime'
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: [Sun Jul 23 2017 13:46:56 GMT-0700 (PDT)] ERROR TypeError: params.then is not a function
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at queryYelp (/home/hubot/mybot/node_modules/hubot-yelp/src/lunchtime.coffee:41:3, <js>:27:31)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at TextListener.callback (/home/hubot/mybot/node_modules/hubot-yelp/src/lunchtime.coffee:63:5, <js>:54:14)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at executeListener (/home/hubot/mybot/node_modules/hubot/src/listener.coffee:65:11, <js>:53:19)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at allDone (/home/hubot/mybot/node_modules/hubot/src/middleware.coffee:44:37, <js>:34:16)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at /home/hubot/mybot/node_modules/async/lib/async.js:274:13
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at Object.async.eachSeries (/home/hubot/mybot/node_modules/async/lib/async.js:142:20)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at Object.async.reduce (/home/hubot/mybot/node_modules/async/lib/async.js:268:15)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at /home/hubot/mybot/node_modules/hubot/src/middleware.coffee:49:7, <js>:37:22
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at _combinedTickCallback (internal/process/next_tick.js:73:7)
Jul 23 13:46:56 n7-z01-0a2a1068 hubot: at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:128:9)
```

I tried putting `params` in parentheses and it seems to be happier that way (no errors), but it seems like the messages came in out of order:
![image](https://user-images.githubusercontent.com/18687001/28502778-e0565eae-6fc7-11e7-9407-f9651bcf3add.png)

Would it be possible to put those all in a single message before sending it?
Thanks!

**Comment by jvarness** [comment]
That's quite strange. I don't know why it worked when I tested it and then it err'd out when others are trying it.

I'll fix this and #15 at the same time.

**Comment by iancward** [comment]
What version of NodeJS and coffeescript are you using? Maybe mine is different.
I'm using nodejs-6-11.1 and coffeescript version 1.6.3.
